src/com/facebook/buck/features/apple/project/SchemeGenerator.java

```diff
@@ -387,6 +399,14 @@ public static Element serializeEnvironmentVariables(
     return rootElement;
   }
 
+  public static Element serializeExpandVariablesBasedOn(
+    Document doc, XCScheme.BuildableReference reference) {
+    Element referenceElement = serializeBuildableReference(doc, reference);
+    Element rootElement = doc.createElement("MacroExpansion");
```

Comment:
Is this what you mentioned reverse-engineered? "MacroExpansion". It should be Ok because Apple uses it in their open source https://opensource.apple.com/source/Security/Security-57337.20.44/Security.xcodeproj/xcshareddata/xcschemes/Debug.xcscheme.auto.html

Reply:
Yes, I didn't found documentation about the format, so by reverse-engineered I mean that I have checked schemas generated by Xcode (format + behaviour for different actions).